UMLEditor/UMLEditor/Classes/AutoCompleteHandler.cs

```diff
@@ -0,0 +1,100 @@
+﻿using System;
+using System.Collections.Generic;
+
+namespace UMLEditor.Classes;
+
+/// <summary>
+/// 
+/// </summary>
+public class AutoCompleteHandler : IAutoCompleteHandler
+{
+    /// <summary>
+    /// Separate different parts of list
+    /// </summary>
+    public char[] Separators { get; set; } = new char[] { ' ' };
+
+    /// <summary>
+    /// List of commands supported
+    /// </summary>
+    public string[] SupportedCommands { get; set; } = Array.Empty<string>();
+
+    /// <summary>
+    /// Gets suggestions given what user has typed
+    /// </summary>
+    /// <param name="index">index of where user typed up to in string</param>
+    /// <param name="forLine">Line to attempt to complete</param>
+    /// <returns></returns>
+    public string[] GetSuggestions(int index, string forLine)
+    {
+        List<string> candidates = new List<string>();
+        foreach (var command in SupportedCommands)
+        {
+            if (IsLikelyCandidate(forLine, command))
+            {
+                candidates.Add(command);
+            }
+        }
+
+        return candidates.ToArray();
+    }
+
+    /// <summary>
+    /// Returns the last matching index of a string
+    /// </summary>
+    /// <param name="first">The first string</param>
+    /// <param name="second">The second string</param>
+    /// <returns>The last matching index</returns>
+    private int LastMatchingIndex(string first, string second)
+    {
+        int result = -1;
+
+        int lastIndex = Math.Min(first.Length, second.Length);
+        for (int index = 0; index < lastIndex; ++index)
+        {
+            if (first[index] == second[index])
+            {
+                ++result;
+            }
+            else
+            {
+                break;
+            }
+        }
+
+        return result;
+    }
```

Comment:
This seems to be using my initial demo implementation, which was a rough proof of concept. A cleaner way to do canidate finding would be to use the index provided in LastMatchinIndex to do size comparisions and String.StartsWith to see if the command is a match. 

For instance, if the user entered 10 characters and a command is 5 you immediately know it isn't a match. If the number of characters is <= the length of the command then if the command starts with the entered text it is a match.
Commands: add_class add_method
User entered add (4 characters)
Both start with add_: **add**_class **add**_method